cmd/access/unstaked_access_node_builder.go

```diff
@@ -122,6 +142,51 @@ func (builder *UnstakedAccessNodeBuilder) enqueueUnstakedNetworkInit() {
 
 		builder.Logger.Info().Msgf("unstaked network will run on address: %s", builder.unstakedNetworkBindAddr)
 
+		builder.upstreamConnector = &UpstreamConnector{
+			middleware: middleware,
+			logger:     builder.Logger,
+			stakedAN:   modifiedStakedAN,
+		}
+
 		return builder.UnstakedNetwork, err
 	})
 }
+
+func (builder *UnstakedAccessNodeBuilder) enqueueConnectUpstream() {
+	builder.Component("connect upstream", func(_ cmd.NodeBuilder, node *cmd.NodeConfig) (module.ReadyDoneAware, error) {
+		return builder.upstreamConnector, nil
+	})
+}
+
+type UpstreamConnector struct {
+	middleware *p2p.Middleware
+	stakedAN   *flow.Identity
+	logger     zerolog.Logger
+}
+
+func (connector *UpstreamConnector) Ready() <-chan struct{} {
```

Comment:
This might be okay for now, but we probably want to refactor this to make it concurrecy-safe / idempotent.

See https://github.com/dapperlabs/flow-go/issues/5702.

This should be done easily once https://github.com/onflow/flow-go/pull/1031 is merged.